modules/tb-all-in-one/user_data.sh

```diff
@@ -1,21 +1,40 @@
-#!/bin/bash
+#!/bin/bash -ex
 set -euo pipefail
 
+exec > >(tee /var/log/user-data.log|logger -t user-data -s 2>/dev/console) 2>&1
+
+echo BEGIN
+
+echo "Setting up ansible things"
+
 # Install basic stuff for ansible
 apt-get update
 apt-get -y install python-simplejson python-pip libpq-dev
+pip install --upgrade pip
 pip install psycopg2
 
+echo "Setting up disk mount"
+
 # Setup the DO Volume, formatting it if it is unformatted
 # and then mounting it
 DISK=/dev/disk/by-id/scsi-0DO_Volume_${name}
 MOUNT=/mnt/${name}
 
-DEVICE_FS=`blkid -o value -s TYPE $DISK`
+echo "Checking disk format"
+# Echo is so we don't exit the script because blkid is stupid and returns
+# exit code 2 when the disk isn't formatted >:(
+DEVICE_FS=`blkid -o value -s TYPE $DISK || echo ""`
+echo "Disk is formatted? $DEVICE_FS"
 if [ "`echo -n $DEVICE_FS`" == "" ] ; then
+  echo "Formatting $DISK"
```

Comment:
uh you're not supposed to script things like this. 

sources of fire, sources of destruction, and sources of filesystem shouldn't be automated...

Reply:
Yolo.